### ISSUE TITLE
[Ionic v4] Freezed navigation with children routes (with or wihout multiple nested ion-router-outlet)

### ISSUE THREAD

**Issue by giacomocerquone** [ionic-team/ionic#15959]
# Bug Report

**Ionic Info**

```
✔ Gathering environment info - done!

Ionic:

   ionic (Ionic CLI)             : 4.2.1 (/Users/giacomocerquone/.nvm/versions/node/v10.3.0/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.0.0-beta.12
   @angular-devkit/build-angular : 0.7.5
   @angular-devkit/schematics    : 0.7.5
   @angular/cli                  : 6.1.5
   @ionic/angular-toolkit        : 1.0.0

System:

   NodeJS : v10.3.0 (/Users/giacomocerquone/.nvm/versions/node/v10.3.0/bin/node)
   npm    : 6.4.0
   OS     : macOS High Sierra
```

**Describe the Bug**

This is an issue made to collect and further expand some of the weird behaviours of the new navigation system in ionic 4, that, I believe, are referable to this one bug.
More precisely a bug in the ion-router-outlet component (or directive, as you wish).
I took some time to shoot a little video where I explain the problem in detail, showing the code and the app running (sorry if it's not up to the standards, but it's my first video): https://youtu.be/q5lWTb1gT9k

**Related Code**
There is this repo that you can examine: https://github.com/giacomocerquone/ionicv4-routing-bug.git

**Expected Behavior**
That I could experience an usual angular navigation in my app. 

**Additional Context**

This is, again, the video I made: https://youtu.be/q5lWTb1gT9k
These are all the issues I could track down that are surely related to the same bug inside the outlet:

- #15370 
- #14888 
- #15798 
- #15449 
- #15295 

I'm doing this since I really hope to see this bug fixed that I find very critical. I'm about to ship a production app (to deliver for the next month) and I don't know how to fix this situation.

If some of you knows a "quick fix" to apply in the meantime that they fix this problem, it's all ears!!!
Thanks so much everyone :)

**Comment by louis-raine** [comment]
+ 1

**Comment by sachst1000** [comment]
Same issue

**Comment by sachst1000** [comment]
Moving from one module to another via router.navigate([SomeURL]) works for the first view.  Subsequent router.navigate([...]) within next module (calling other views) just sit on the first view.  This is killing us.  Any time frame as to when this can be fixed?

**Comment by Silvest89** [comment]
This still happens even with absolute urls especially when making use of back buttons and goBack() in rapid succession.

**Comment by Silvest89** [comment]
> @Silvest89 no. I assure you that, with absolute links, it does work! I'm using those links in the router.navigate method

How would that work in conjunction with the hardware back button ?:P
And back buttons for one page that can differ depending on from the page you were before you entered the page. I'd probably have to use a service or a method to keep track until it has been fixed xD

**Comment by TiVoMaker** [comment]
See: https://github.com/angular/angular/issues/27050

The back-button  uses router.navigateByUrl(), which creates a new UrlSegmentGroup object, thus triggering this bug if relative navigation is attempted afterward. Using absolute paths everywhere works because this code path is never entered.

**Comment by giacomocerquone** [comment]
@TiVoMaker are you sure? Because I created a wrapper around the back button that doesn't use navigatebyurl and the navigation is still broken (I'm using navctrl.goback())

**Comment by TiVoMaker** [comment]
As I understand the Angular code, the bug would be triggered by any navigateByUrl() followed by a relative navigate(). So the first time you land on a page, a relative navigation to some other place will work, however once you hit the back button, the absolute navigation back changes the object reference, and thus the relative navigation fails after that.

You can see this by putting an absolute navigation on the first page that goes somewhere else, and clicking it after you get hung up coming back. Then come back to the first page and try again - it will do the exact same thing. 

I believe you can fix it by implementing an alternative to navigate() that gets the current url (via ActivatedRoute.snapshot.url), then hand-builds an absolute URL using the relative path. You'd do this as a service provider so it can get the ActivatedRoute and Router. I'm testing my own version of this right now :-)

**Comment by TiVoMaker** [comment]
This seems to work for me:

https://gist.github.com/TiVoMaker/825b48ed79f871adbf588eb27e59c126

**Comment by urbanjoe** [comment]
Same issue....

**Comment by liamdebeasi** [comment]
Hi everyone,

This issue has been fixed with the release of Ionic 4.2.0. Please update to that version, and let me know if you are still experiencing any issues.

Thanks!
